bldsys/cmake/create_gradle_project.cmake

```diff
@@ -33,7 +35,8 @@ set(VALID_ABI "armeabi" "armeabi-v7a" "arm64-v8a" "x86" "x86_64")
 include(${SCRIPT_DIR}/utils.cmake)
 
 # script parameters
-set(ANDROID_API 24 CACHE STRING "")
+set(ANDROID_API 30 CACHE STRING "")
+set(ANDROID_SDK "" CACHE STRING "")
```

Comment:
Is it required to set the ANDROID_SDK environment variable?

Reply:
I might have misunderstood how cmake works here. but its not setting 'environment variable' here. Its just a local variable that I have used inside `local.propertise`. Why I didn't just use `ANDROID_HOME` in local.propertise is because I don't know how to cmake `config file` using enviroment variables. I think it only works with local variables. Someone can correct me if I am wrong.

Reply:
It is just a local variable being used here.  You'd need ENV to set it to an environment variable.  So you can do it with this: https://cmake.org/cmake/help/latest/variable/ENV.html

Reply:
Not sure I follow. My problem is I don't know how to do a cmake `config_file` with environment variables. NOT how to reach ENV var in a cmake script.

At the moment I am doing
```
sdk.dir=@ANDROID_SDK@
```
Are you saying something like the following works?
```
sdk.dir=@$ENV(ANDROID_HOME)@
```

Reply:
However, as there is a local.properties.in this is the way to do a config file:

local.properties.in:
`sdk.dir="${ANDROID_SDK}"`

create_gradle_project.cmake:
`
set(ENV ANDROID_SDK "" CACHE STRING "")
configure_file(local.properties.in local.properties)
`

So multiple ways to achieve your desired results.  If you do the configure_file version, there's no need to add to the environment vars so a simple set(ANDROID_SDK "") is enough to make things happy.

Reply:
Actually `sdk.dir=$ENV{ANDROID_HOME}` would work and create a valid `local.properties` But the problem is I can't do any sanity checks on `ANDROID_HOME` in the `local.properties.in` its better done in `CMakelists.txt` as I am doing now.

Reply:
Also, can I recommend we use android.ndkPath in the projects gradle?  As of AS4.1, this would be the way to tell gradle where to find the NDK.
We could easily just place ndk.dir=blah blah blah in the local.properties.in file.  However, this is deprecated in AS4.1 (while it still works and might be cleaner as this PR adds local.properties generation at the cmake invocation.
Either way, if we define the path to the ndk, we should be able to reliably not require AndroidStudio even be opened at any point in the build process.  Also it will fix the issues reported with regards to not finding the NDK on project open.

Reply:
I thought ndk has to be inside your sdk these days. This is how its finding it now. Originally I had ndk outside my sdk, multiple copies of it actually. If you setup newly these days, installing SDK with ndk component will install it inside it. So don't need to set `android.ndkPath` in the future.

Reply:
It doesn't *have* to be inside the SDK.  This is just very common.  android.version DSL is usually used to allow for gradle to do the right thing.  However, it is still a thing (I do it on CICD instances) to install the SDK without Android studio and the NDK can be installed anywhere.  It's also a setup popular in older game engine environments that haven't really been updated to the latest DSL in a while.
I usually suggest being explicit in android project creation in such a way as to where AS does the right thing whether it generates anything or not (i.e. all projects should build without AS even being installed).
One would just need the SDK+NDK.  AS is just a way to package it and make it easier to work with for developers.

Reply:
I never use AS to install SDK or NDK. I always do it via the command line tools. And AFAICT this is their default behaviour now.